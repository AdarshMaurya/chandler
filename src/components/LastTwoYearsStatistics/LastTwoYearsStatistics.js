import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {View, Text, Image} from 'react-native';

// import {commonStyles} from 'theme';
import {styles} from './LastTwoYearsStatistics.styles';

export class LastTwoYearsStatistics extends PureComponent {
  static propTypes = {
    image: PropTypes.number.isRequired,
    count1: PropTypes.number.isRequired,
    title1: PropTypes.string.isRequired,
    count2: PropTypes.number.isRequired,
    title2: PropTypes.string.isRequired,
  };
  render() {
    const {image, count1, count2, title1, title2} = this.props;

    return (
      <View style={styles.container}>
        <Image source={image} style={styles.image} />
        <View style={styles.leftContainer}>
          <Text style={styles.count}>{count1}</Text>
          <Text style={styles.text}>{title1}</Text>
        </View>
        <View style={styles.rightContainer}>
          <Text style={styles.count}>{count2}</Text>
          <Text style={styles.text}>{title2}</Text>
        </View>
      </View>
    );
  }
}
