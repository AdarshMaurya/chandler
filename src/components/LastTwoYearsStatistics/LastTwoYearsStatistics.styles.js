import {PixelRatio, StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  container: {
    marginVertical: 15,
    flexDirection: 'row',
    alignItems: 'center',
    borderRadius: 3,
    borderColor: 'grey',
    borderWidth: 1 / PixelRatio.get(),
  },
  leftContainer: {
    paddingHorizontal: 10,
    paddingVertical: 7,
    flex: 1,
  },
  rightContainer: {
    paddingHorizontal: 10,
    paddingVertical: 7,
    flex: 1,
    borderColor: 'grey',
    borderLeftWidth: 1 / PixelRatio.get(),
  },
  image: {
    marginHorizontal: 10,
  },
  count: {
    fontSize: 25,
  },
  text: {
    color: '#355974',
  },
});
