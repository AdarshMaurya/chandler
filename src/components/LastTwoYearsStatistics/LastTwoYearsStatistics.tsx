import React, {PureComponent} from 'react';
import {Image, Text, View} from 'react-native';

// import {commonStyles} from 'theme';
import {styles} from './LastTwoYearsStatistics.styles';

interface ILastTwoYearsStatisticsProps {
  image: number | {uri: string};
  count1: number;
  title1: string;
  count2: number;
  title2: string;
}

export class LastTwoYearsStatistics extends PureComponent<
  ILastTwoYearsStatisticsProps
> {
  public render() {
    const {image, count1, count2, title1, title2} = this.props;

    return (
      <View style={styles.container}>
        <Image source={image} style={styles.image} />
        <View style={styles.leftContainer}>
          <Text style={styles.count}>{count1}</Text>
          <Text style={styles.text}>{title1}</Text>
        </View>
        <View style={styles.rightContainer}>
          <Text style={styles.count}>{count2}</Text>
          <Text style={styles.text}>{title2}</Text>
        </View>
      </View>
    );
  }
}
