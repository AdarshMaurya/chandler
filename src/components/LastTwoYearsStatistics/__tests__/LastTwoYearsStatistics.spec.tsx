import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import {LastTwoYearsStatistics} from '../LastTwoYearsStatistics';

describe('Components', () => {
  describe('LastTwoYearsStatistics', () => {
    it('should renders correctly', () => {
      expect(
        toJson(
          shallow(
            <LastTwoYearsStatistics
              title1="title1"
              title2="title2"
              count1={1}
              count2={3}
              image={3}
            />
          )
        )
      ).toMatchSnapshot();
    });
  });
});
