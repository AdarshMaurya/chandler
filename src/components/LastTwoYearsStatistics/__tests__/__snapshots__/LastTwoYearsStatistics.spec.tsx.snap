// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components LastTwoYearsStatistics should renders correctly 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "borderColor": "grey",
      "borderRadius": 3,
      "borderWidth": 0.5,
      "flexDirection": "row",
      "marginVertical": 15,
    }
  }
>
  <Image
    source={3}
    style={
      Object {
        "marginHorizontal": 10,
      }
    }
  />
  <Component
    style={
      Object {
        "flex": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 7,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 25,
        }
      }
    >
      1
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#355974",
        }
      }
    >
      title1
    </Text>
  </Component>
  <Component
    style={
      Object {
        "borderColor": "grey",
        "borderLeftWidth": 0.5,
        "flex": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 7,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 25,
        }
      }
    >
      3
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#355974",
        }
      }
    >
      title2
    </Text>
  </Component>
</Component>
`;
